### ISSUE TITLE
[🐛 Bug]: Passing args as an array to the Appium service creates erroneous arguments 

### ISSUE THREAD

**Issue by TheMagzuz** [webdriverio/webdriverio#10942]
### Have you read the Contributing Guidelines on issues?

- [X] I have read the [Contributing Guidelines on issues](https://github.com/webdriverio/webdriverio/blob/main/CONTRIBUTING.md#reporting-new-issues).

### WebdriverIO Version

8.12.1

### Node.js Version

v18.16.1

### Mode

WDIO Testrunner

### What happened?

When using the Appium service, passing the `args` as an array, as shown in the [documentation](https://webdriver.io/docs/appium-service/#args) causes Appium to be started with indices before each argument, e.g., the configuration
```js
{
//...
    services: [['appium', {
        args: [
            '--allow-insecure=adb_shell',
        ],
    },
    ]],
//...
}
```
starts Appium with the arguments: `node [APPIUM PATH] --0 --allow-insecure=adb_shell --base-path /`

### What is your expected behavior?

As stated in the documentation, passing `args` as an array should result in each element in the array becoming an argument.

### How to reproduce the bug.

1. Create a new project with `npm init wdio .`, selecting the Appium service to be included
2.  In `wdio.conf.ts`, replace the `services` field with:
```js
    services: [['appium', {
        args: [
            '--allow-insecure=adb_shell',
        ],
    },
    ]],
```
3.  Run the project with `npm run wdio`. The project will attempt to start Appium with the arguments `--0 --allow-insecure=adb_shell --base-path /`, and fail since the argument `--0` is not valid

### Relevant log output

```typescript
> wdio-test@1.0.0 wdio
> wdio run ./wdio.conf.ts


Execution of 1 workers started at 2023-08-14T13:05:25.293Z

2023-08-14T13:05:25.307Z INFO @wdio/cli:launcher: Run onPrepare hook
2023-08-14T13:05:25.308Z INFO @wdio/appium-service: Will spawn Appium process: node /home/markus/wdio-test/node_modules/appium/index.js --0 --allow-insecure=adb_shell --base-path /
2023-08-14T13:05:26.916Z ERROR @wdio/cli:utils: A service failed in the 'onPrepare' hook
Error: Appium exited before timeout (exit code: 1)
    at ChildProcess.<anonymous> (file:///home/markus/wdio-test/node_modules/@wdio/appium-service/build/launcher.js:115:22)
    at Object.onceWrapper (node:events:628:26)
    at ChildProcess.emit (node:events:513:28)
    at ChildProcess.emit (node:domain:489:12)
    at Process.ChildProcess._handle.onexit (node:internal/child_process:291:12)

Continue...
2023-08-14T13:05:26.918Z INFO @wdio/cli:launcher: Run onWorkerStart hook
2023-08-14T13:05:26.918Z INFO @wdio/local-runner: Start worker 0-0 with arg: run,./wdio.conf.ts
[0-0] 2023-08-14T13:05:27.842Z INFO @wdio/local-runner: Run worker command: run
[0-0] RUNNING in Chrome - file:///test/specs/test.e2e.ts
[0-0] 2023-08-14T13:05:27.994Z INFO webdriver: Initiate new session using the WebDriver protocol
[0-0] 2023-08-14T13:05:27.995Z INFO webdriver: Connecting to existing driver at http://127.0.0.1:4723/
[0-0] 2023-08-14T13:05:27.998Z INFO webdriver: [POST] http://127.0.0.1:4723/session
[0-0] 2023-08-14T13:05:27.998Z INFO webdriver: DATA {
[0-0]   capabilities: {
[0-0]     alwaysMatch: {
[0-0]       platformName: 'Android',
[0-0]       browserName: 'Chrome',
[0-0]       'appium:deviceName': 'Android GoogleAPI Emulator',
[0-0]       'appium:platformVersion': '12.0',
[0-0]       'appium:automationName': 'UiAutomator2'
[0-0]     },
[0-0]     firstMatch: [ {} ]
[0-0]   },
[0-0]   desiredCapabilities: {
[0-0]     platformName: 'Android',
[0-0]     browserName: 'Chrome',
[0-0]     'appium:deviceName': 'Android GoogleAPI Emulator',
[0-0]     'appium:platformVersion': '12.0',
[0-0]     'appium:automationName': 'UiAutomator2'
[0-0]   }
[0-0] }
[0-0] 2023-08-14T13:05:28.008Z ERROR webdriver: RequestError: connect ECONNREFUSED 127.0.0.1:4723
[0-0]     at ClientRequest.<anonymous> (file:///home/markus/wdio-test/node_modules/got/dist/source/core/index.js:790:107)
[0-0]     at Object.onceWrapper (node:events:628:26)
[0-0]     at ClientRequest.emit (node:events:525:35)
[0-0]     at ClientRequest.emit (node:domain:489:12)
[0-0]     at Socket.socketErrorListener (node:_http_client:502:9)
[0-0]     at Socket.emit (node:events:513:28)
[0-0]     at Socket.emit (node:domain:489:12)
[0-0]     at emitErrorNT (node:internal/streams/destroy:151:8)
[0-0]     at emitErrorCloseNT (node:internal/streams/destroy:116:3)
[0-0]     at processTicksAndRejections (node:internal/process/task_queues:82:21)
[0-0]     at TCPConnectWrap.afterConnect [as oncomplete] (node:net:1494:16)
[0-0] 2023-08-14T13:05:28.008Z ERROR @wdio/runner: Error: Failed to create session.
[0-0] Unable to connect to "http://127.0.0.1:4723/", make sure browser driver is running on that address.
[0-0] It seems like the service failed to start or is rejecting any connections.
[0-0]     at startWebDriverSession (file:///home/markus/wdio-test/node_modules/webdriver/build/utils.js:68:15)
[0-0]     at processTicksAndRejections (node:internal/process/task_queues:95:5)
[0-0]     at async Function.newSession (file:///home/markus/wdio-test/node_modules/webdriver/build/index.js:28:45)
[0-0]     at async remote (file:///home/markus/wdio-test/node_modules/webdriverio/build/index.js:45:22)
[0-0]     at async Runner._startSession (file:///home/markus/wdio-test/node_modules/@wdio/runner/build/index.js:222:29)
[0-0]     at async Runner._initSession (file:///home/markus/wdio-test/node_modules/@wdio/runner/build/index.js:188:25)
[0-0]     at async Runner.run (file:///home/markus/wdio-test/node_modules/@wdio/runner/build/index.js:77:19)
[0-0] FAILED in Chrome - file:///test/specs/test.e2e.ts
2023-08-14T13:05:28.171Z INFO @wdio/cli:launcher: Run onWorkerEnd hook
2023-08-14T13:05:28.172Z INFO @wdio/cli:launcher: Run onComplete hook

Spec Files:	0 passed, 1 failed, 1 total (100% completed) in 00:00:02 

2023-08-14T13:05:28.173Z INFO @wdio/local-runner: Shutting down spawned worker
2023-08-14T13:05:28.426Z INFO @wdio/local-runner: Waiting for 0 to shut down gracefully
2023-08-14T13:05:28.427Z INFO @wdio/local-runner: shutting down
```

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### Is there an existing issue for this?

- [X] I have searched the existing issues

**Comment by christian-bromann** [comment]
Thanks for reporting!

We greatly appreciate any contributions that help resolve the bug. While we understand that active contributors have their own priorities, we kindly request your assistance if you rely on this bug being fixed. We encourage you to take a look at our [contribution guidelines](https://webdriver.io/docs/contribute) or join our friendly [Discord](http://discord.webdriver.io/) development server, where you can ask any questions you may have. Thank you for your support, and cheers!

**Comment by christian-bromann** [comment]
Thanks @tamil777selvan for bringing light into this issue.

@TheMagzuz please see #10972, we don't support passing args as array to the Appium server. We recommend using the following structure:

```ts
    services: [['appium', {
        args: {
            allowInsecure: 'adb_shell',
            // ...
        },
    }]],
```

Let me know if you have further questions.

**Comment by Anandhakrishnansm** [comment]
@christian-bromann I am also facing the issue.
I tried with below changes in my wdio file but still I am getting some errors while running it.

Changes I made in wdio : 
![image](https://github.com/webdriverio/webdriverio/assets/102238148/fe5bb61f-12b8-426c-8338-ce9e237fd10d)

Issue : 
`yarn run v1.22.19
$ wdio run ./wdio.conf.js

Execution of 1 workers started at 2023-11-20T10:14:42.895Z

2023-11-20T10:14:42.957Z DEBUG @wdio/utils:initializeServices: initialize service "appium" as NPM package
2023-11-20T10:14:43.460Z INFO @wdio/local-runner: Shutting down spawned worker
2023-11-20T10:14:43.717Z INFO @wdio/local-runner: Waiting for 0 to shut down gracefully
2023-11-20T10:14:43.718Z INFO @wdio/local-runner: shutting down
TypeError: TypeError: this.runner.initialize is not a function
    at Launcher.run (file:///Users/builder/clone/node_modules/@wdio/cli/build/launcher.js:92:31)
TypeError: this.runner.initialize is not a function
    at Launcher.run (file:///Users/builder/clone/node_modules/@wdio/cli/build/launcher.js:92:31)
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.


Build failed :|
Step 8 script `Run WebDriver test suite` exited with status code 1`
